generators/app/templates/src/app/reducer.ts

```diff
@@ -0,0 +1,48 @@
+import { Nullable } from '~utils/types';
+
+export interface User {
+  id: number;
+}
+
+export interface GlobalState {
+  user: Nullable<User>;
+}
+
+export const INITIAL_STATE = {
+  user: null
+};
+
+enum ActionTypes {
+  SET_USER = 'SET_USER',
+  RESET_USER = 'RESET_USER'
+}
+
+export interface SetUser {
+  type: ActionTypes.SET_USER;
+  payload: User;
+}
+
+export interface ResetUser {
+  type: ActionTypes.RESET_USER;
+}
+
+export type Action = SetUser | ResetUser;
+
+export const actionCreators = {
+  setUser: (user: User) => ({ type: ActionTypes.SET_USER, payload: user }),
+  resetUser: () => ({ type: ActionTypes.RESET_USER })
+};
+
+export const globalReducer = (state: GlobalState, action: Action): GlobalState => {
```

Comment:
I don't really get what this is for. Is it for handling user data? Shouldn't that have a different name?

Reply:
We'll now have more than one store:
1) Each screen will have its store
2) A Global store to save stuff that we'll need to keep from screen to screen, which is normally the user, app theme of something like that. It's not limited to user data, but I think it'll usually be that.

Reply:
It's a good point.
I'm just thinking out loud, but wouldn't it make sense to have different contexts for different global data?
To avoid unnecessary updates, for example updating consumers of the user when the theme changes.